homeassistant/components/yeelight/light.py

```diff
@@ -609,7 +609,10 @@ def _get_hs_from_properties(self):
     def set_music_mode(self, music_mode) -> None:
         """Set the music mode on or off."""
         if music_mode:
-            self._bulb.start_music()
+            try:
+                self._bulb.start_music()
+            except AssertionError as ex:
```

Comment:
Why do we rely on assert in production code?

Reply:
Assertion error is raised in python-yeelight lib: https://gitlab.com/stavros/python-yeelight/-/blob/master/yeelight/main.py#L807 . Looks like its the only case, when its raised. All other method seems to stick with BulbException.

Reply:
Ok. It would be better if the library only used library specific exceptions. 🤷‍♂️